MapboxNavigation/RatingControl.swift

```diff
@@ -0,0 +1,135 @@
+        }
+    }
+    
+    //MARK: Initializers
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        configureStars()
+    }
+    
+    required init(coder: NSCoder) {
+        super.init(coder: coder)
+        configureStars()
+    }
+    
+    //MARK: Private Functions
+    private func configureStars() {
+        removeStars()
+        addStars()
+        updateSelectionStates()
+    }
+    
+    private func addStars() {
+        for index in 0..<starCount {
+            let button = UIButton(type: .custom)
+            button.setImage(starTemplate, for: .normal)
+            button.adjustsImageWhenHighlighted = false
+            addButtonSizeConstraints(to: button)
+            
+            let setRatingNumber = NSNumber(value: index + 1)
+            let setRatingString = NumberFormatter.localizedString(from: setRatingNumber, number: .none)
+            let localizedStarLabel = NSLocalizedString("Set %@ star rating", comment: "Accessibility Star Label")
+            button.accessibilityLabel = String.localizedStringWithFormat(localizedStarLabel, setRatingString)
+                
+            
+            button.addTarget(self, action: #selector(RatingControl.ratingButtonTapped(button:)), for: .touchUpInside)
+            
+            addArrangedSubview(button)
+            
+            stars.append(button)
+        }
+    }
+    
+    private func removeStars() {
+        stars.forEach {
+            removeArrangedSubview($0)
+            $0.removeFromSuperview()
+        }
+        stars.removeAll()
+    }
+    
+    private func updateSelectionStates() {
+        for (index, button) in stars.enumerated() {
+            
+            let selected = index < rating
+            button.tintColor = selected  ? selectedColor : normalColor
+            button.isSelected = selected
+            
+            setAccessibility(for: button, at: index)
+        }
+    }
+    
+    private func setAccessibility(for button: UIButton, at index: Int) {
+        setAccessibilityHint(for: button, at: index)
+        
+        let value: String
+        
+        switch rating {
+        case 0:
+            value = "No rating set."
+        case 1:
+            value = "1 star set."
+        default:
+            value = "\(rating) stars set."
+        }
+        
+        button.accessibilityValue = value
+    }
+    
+    private func setAccessibilityHint(for button: UIButton, at index: Int) {
+        guard rating == (index + 1) else { return } //This applies only to the zero-resettable button.
+        
+        button.accessibilityHint = NSLocalizedString("Tap to reset the rating to zero.", comment: "Rating Reset To Zero Accessability Hint")
+    }
+    
+    
+    private func addButtonSizeConstraints(to view: UIView) {
+        let width = NSLayoutConstraint(item: view, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1.0, constant: starSize.width)
```

Comment:
Nit: Since we're requiring iOS >= 9, we can use the fluent API. This can be written more succinctly as:
```
view.widthAnchor.constraint(equalToConstant: starSize.width).isActive = true
```

Reply:
- [ ] Use Layout Anchors